src/client/camera.cpp

```diff
@@ -706,19 +707,20 @@ void Camera::drawNametags()
 		if (transformed_pos[3] > 0) {
 			std::wstring nametag_colorless =
 				unescape_translate(utf8_to_wide(nametag->nametag_text));
-			core::dimension2d<u32> textsize =
-				g_fontengine->getFont()->getDimension(
+			core::dimension2d<u32> textsize = font->getDimension(
 				nametag_colorless.c_str());
 			f32 zDiv = transformed_pos[3] == 0.0f ? 1.0f :
 				core::reciprocal(transformed_pos[3]);
-			v2u32 screensize = RenderingEngine::get_video_driver()->getScreenSize();
 			v2s32 screen_pos;
 			screen_pos.X = screensize.X *
 				(0.5 * transformed_pos[0] * zDiv + 0.5) - textsize.Width / 2;
 			screen_pos.Y = screensize.Y *
 				(0.5 - transformed_pos[1] * zDiv * 0.5) - textsize.Height / 2;
 			core::rect<s32> size(0, 0, textsize.Width, textsize.Height);
-			g_fontengine->getFont()->draw(
+			core::rect<s32> bg_size(-2, 0, textsize.Width+2, textsize.Height);
+			driver->draw2DRectangle(video::SColor(50, 0, 0, 0),
```

Comment:
The color should not be hardcoded like this: What if a nametag happens to be grey? Will it be legible?